esm/utils.js

```diff
@@ -110,16 +110,20 @@ export const importJS = (source, name) => import(source).then(esm => {
 });
 
 export const importCSS = href => new Promise((onload, onerror) => {
-    if (document.querySelector(`link[href="${href}"]`)) onload();
-    document.head.append(
-        assign(
-            document.createElement('link'),
-            { rel: 'stylesheet', href, onload, onerror },
-        )
-    )
+    if (document.querySelector(`link[rel="stylesheet"][href="${href}"]`)) {
+        onload();
+    }
+    else {
+        document.head.append(
+            assign(
+                document.createElement('link'),
+                { rel: 'stylesheet', href, onload, onerror },
+            )
+        );
+    }
 });
 
-export const isCSS = source => /\.css/i.test(new URL(source).pathname);
+export const isCSS = source => /\.css$/i.test(new URL(source).pathname);
```

Comment:
What is the purpose of the `$` here..?

Reply:
this is the equivalent of `new URL(source).pathname.endsWith('.css')` ... the `pathname` never carries `?qs` or `#hash` values so before a file named `stuff.css.js` would've returned `true`, now it doesn't.

```js
/\.css/i.test('file.css.js');  // true
/\.css$/i.test('file.css.js'); // false
```

Reply:
arguably it was a sneaky fix within this MR but it's all good (imho).